enterprise/suppress-user/factory.go

```diff
@@ -0,0 +1,82 @@
+package suppression
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/config"
+	backendconfig "github.com/rudderlabs/rudder-server/config/backend-config"
+	"github.com/rudderlabs/rudder-server/rruntime"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+	"github.com/rudderlabs/rudder-server/utils/types"
+)
+
+type Factory struct {
+	EnterpriseToken string
+}
+
+// New initializes Suppress User feature
+func (m *Factory) Setup(ctx context.Context, backendConfig backendconfig.BackendConfig) (types.UserSuppression, error) {
+	log := logger.NewLogger().Child("enterprise").Child("suppress-user")
+	if m.EnterpriseToken == "" {
+		log.Info("Suppress User feature is enterprise only")
+		return &NOOP{}, nil
+	}
+	log.Info("Setting up Suppress User Feature")
+	backendConfig.WaitForConfig(ctx)
+	var repository Repository
+	if config.GetBool("BackendConfig.Regulations.useBadgerDB", true) {
+		tmpDir, err := misc.CreateTMPDIR()
+		if err != nil {
+			return nil, fmt.Errorf("could not create tmp dir: %w", err)
+		}
+		path := path.Join(tmpDir, "suppression")
+
+		// TODO: implement seeder source, to retrieve the initial state from a persisted backup
+		var seederSource func() (io.Reader, error)
+
+		repository, err = NewBadgerRepository(
+			path,
+			log,
+			WithSeederSource(seederSource),
+			WithMaxSeedWait(config.GetDuration("BackendConfig.Regulations.maxSeedWait", 5, time.Second)))
```

Comment:
**Explain:** if restore doesn't complete in 5 seconds, the gateway container will startup. Only after restore completes will the gateway be able to successfully perform suppressions, in the meantime, no suppressions will be performed.